app/models/KafkaMessageReader.scala

```diff
@@ -0,0 +1,109 @@
+package models
+
+import java.io.{ByteArrayInputStream, InputStreamReader}
+import java.util.zip.{GZIPInputStream, ZipException}
+
+import com.google.common.io.CharStreams
+import kafka.api.{FetchRequestBuilder, PartitionMetadata, TopicMetadataRequest}
+import kafka.consumer.SimpleConsumer
+import play.api.Play.current
+import play.api.{Logger, Play}
+
+/**
+  * Contains code to read messages from kafka.
+  */
+object KafkaMessageReader {
+
+  private val Kafkas = Play.configuration.getString("capillary.kafkas").getOrElse("127.0.0.1:9092")
+  private val KafkaHosts = Kafkas.split(",")
+
+  private val Timeout = 100000
+  private val BufferSize = 10000000
+  private val ClientId = "Capillary"
+
+  def getMessageContent(topic: String, partition: Int, offset: Long): String = {
+    findLeader(KafkaHosts, topic, partition) match {
+      case Some(pm) => readMessage(pm, topic, offset)
+      case None => s"Could not find any leader for partition: $partition"
+    }
+  }
+
+  private def findLeader(kafkaHosts: Array[String], topic: String, partition: Int): Option[PartitionMetadata] = {
+    var simpleConsumer: Option[SimpleConsumer] = None
```

Comment:
I'm not totally enamored with the mutable approach, especially since I think there's at least one mutation-related bug present.  Here's a possible way to model it purely (NB: this is only psuedocode, but I'm happy to pair to make it functional if wanted):

```
val pm: Option[PartitionMetadata] = kafkaHosts
  .view
  .map(new SimpleConsumer(host, port, Timeout, BufferSize, ClientId))
  .map(mapToPartitionMetadata)
  .flatten
  .find(_.partitionId == partition)

def mapToPartitionMetadata(consumer: SimpleConsumer): PartitionMetadata = try {
  val topicMetadataRequest = new TopicMetadataRequest(Seq(topic), 0)
  val topicMetadataResponse = simpleConsumer.get.send(topicMetadataRequest)
  for (topicMetadata <- topicMetadataResponse.topicsMetadata) {
  val metadata = topicMetadata.partitionsMetadata) {
  consumer.close
  metadata
}```

Note the `view` call is there to convert into a lazy sequence, which would only construct the consumer on demand if the previous one failed to satisfy the `find` call.  Thoughts?